apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,851 @@
+
+Example output:
+
+```json
+{
+   "decision": "reject",
+   "message" : "You have exceeded maximum number of MFA attempts."
+}
+```
+**Examples**
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="js"
+  queryGroup="language"
+>
+<TabPanel id="limit-failed-mfa-verification-attempts" label="Limit Failed MFA Verification Attempts">
+
+Imagine that your company requires that a user can input an incorrect MFA Verification code no more than once every 2 seconds.
+
+#### Database Setup
+
+Create a table to record the last time a user had an incorrect MFA verification attempt for a factor.
+
+```sql
+create table public.mfa_failed_verification_attempts (
+  user_id uuid not null,
+  factor_id uuid not null,
+  last_failed_at timestamp not null default now(),
+  primary key (user_id, factor_id)
+);
+```
+
+Create a hook to read and write information to this table. For example:
+
+```sql
+create function public.hook_mfa_verification_attempt(event jsonb)
+  returns jsonb
+  language plpgsql
+as $$
+  declare
+    last_failed_at timestamp;
+  begin
+    if event->'valid' is true then
+      -- code is valid, accept it
+      return jsonb_build_object('decision', 'continue');
+    end if;
+
+    select last_failed_at into last_failed_at
+      from public.mfa_failed_verification_attempts
+      where
+        user_id = event->'user_id'
+          and
+        factor_id = event->'factor_id';
+
+    if last_failed_at is not null and now() - last_failed_at < interval '2 seconds' then
+      -- last attempt was done too quickly
+      return jsonb_build_object(
+        'error', jsonb_build_object(
+          'http_code', 429,
+          'message',   'Please wait a moment before trying again.'
+        )
+      );
+    end if;
+
+    -- record this failed attempt
+    insert into public.mfa_failed_verification_attempts
+      (
+        user_id,
+        factor_id,
+        last_refreshed_at
+      )
+      values
+      (
+        event->'user_id',
+        event->'factor_id',
+        now()
+      )
+      on conflict do update
+        set last_refreshed_at = now();
+
+    -- finally let Supabase Auth do the default behavior for a failed attempt
+    return jsonb_build_object('decision', 'continue');
+  end;
+$$;
+
+-- Assign appropriate permissions and revoke access
+grant execute
+  on function public.hook_mfa_verification_attempt
+  to supabase_auth_admin;
+
+grant all
+  on table public.mfa_failed_verification_attempts
+  to supabase_auth_admin;
+
+revoke execute
+  on function public.hook_mfa_verification_attempt
+  from authenticated, anon;
+
+revoke all
+  on table public.mfa_failed_verification_attempts
+  from authenticated, anon;
+```
+</TabPanel>
+<TabPanel id="send-email-after-failed-attempts" label="Send Email After Failed Verification Attempts">
+
+Enable the `pg_net` extension under [`Database > Extensions`](/dashboard/project/_/database/extensions)
```

Comment:
Maybe add some introduction as to what the example will be doing. Why is pg_net necessary, what service is going to be used to send the email. Adjust the title of the tab accordingly.